packages/compiler/src/render3/partial/component.ts

```diff
@@ -91,8 +91,7 @@ function compileTemplateDefinition(template: ParsedTemplate): o.LiteralMapExpr {
  * individual directives. If the component does not use any directives, then null is returned.
  */
 function compileUsedDirectiveMetadata(meta: R3ComponentMetadata): o.LiteralArrayExpr|null {
-  const wrapType =
-      meta.wrapDirectivesAndPipesInClosure ? generateForwardRef : (expr: o.Expression) => expr;
+  const wrapType = meta.declarationEmitMode ? generateForwardRef : (expr: o.Expression) => expr;
```

Comment:
So this is making the assumption that `DeclarationEmitMode.Direct` is always 0, right?
I think it would be safer to make this explicit:

```suggestion
  const wrapType = meta.declarationEmitMode !== DeclarationEmitMode.Direct ? generateForwardRef : (expr: o.Expression) => expr;
```

SImilarly below.

Reply:
Well spotted! This was changed during a rename refactoring and not intentionally left this way. Fixed for both instances.